test/phoenix/integration/endpoint_test.exs

```diff
@@ -1,21 +1,32 @@
 Code.require_file "../../support/http_client.exs", __DIR__
+Code.require_file "../../support/endpoint_helper.exs", __DIR__
 
 defmodule Phoenix.Integration.EndpointTest do
   use ExUnit.Case
   import ExUnit.CaptureLog
 
+  import Phoenix.Integration.EndpointHelper
+
   alias Phoenix.Integration.AdapterTest.ProdEndpoint
   alias Phoenix.Integration.AdapterTest.DevEndpoint
   alias Phoenix.Integration.AdapterTest.ProdInet6Endpoint
 
+  # Set up the following module variables containing
+  # port numbers for use in the rest of the module.
+  @port_names [:dev, :prod, :prod_inet6]
+
+  for {port_name, port_number} <- Enum.zip(@port_names, get_unused_port_numbers(length(@port_names))) do
+    Module.put_attribute(__MODULE__, port_name, port_number)
+  end
+
```

Comment:
I wanted to continue to allow the tests to use the `@dev` and `@prod` module attributes, but I wasn't able to destructure the results of `get_unused_port_number/1` into module attributes

```elixir
  # Doesn't work
  [@dev, @prod, @prod_inet6] get_unused_port_numbers(3)
```

Any suggestions for a better way to do this would be appreciated. :)